pandas/indexes/multi.py

```diff
@@ -1007,6 +1008,9 @@ def from_tuples(cls, tuples, sortorder=None, names=None):
             # I think this is right? Not quite sure...
             raise TypeError('Cannot infer number of levels from empty list')
 
+        if not _check_equal_length(tuples):
```

Comment:
I would suggest grepping for uses of `to_object_array_tuples` to understand the impact